src/chempotential.jl

```diff
@@ -84,39 +108,77 @@ function log_activity!(
         # прежний код использовал sum( генератор с for in eachindex ) - давал 2 аллокации
         ∂A = 2 * dot(nmol, @view aij[:,i])
 
-        log_a_[i] = -log_VmBbyV + b * ∑molbyVmB
-        brackets = (∂A / A - (c - d)/CmD) * log_VpCbyVpD + (c/VpC - d/VpD)
-        log_a_[i] -= AbyRTCmD * brackets
+        log_ca[i] = -log_VmBbyV + b * ∑molbyVmB
+        brackets = (∂A / A - (c - d) / CmD) * log_VpCbyVpD + (c / VpC - d / VpD)
+        log_ca[i] -= AbyRTCmD * brackets
     end
-    return log_a_
+    return log_ca
+end
+
+"""
+    log_activity_wj(mixture, nmol, volume, RT; buf = (;))
+
+Return vector of ln(c_a) - logarithm of activity coefficient
+for components of `mixture` at given `nmol`, `volume`, `RT` -
+and the jacobian ∂ln(c_a[i]) / ∂n[j].
+If buffers are provided as keyword arguments, their contents
+are modified during the intermediate calculations.
+
+# Arguments
+
+- `mix`: mixture
+- `nmol::AbstractVector`: amount of each component (mol)
+- `volume`: volume of the mixture (m³)
+- `RT`: thermal energy (J/mol)
+
+# Returns
+
+- `AbstractVector`: the logarithms of activity coefficients
+of the components at given number of moles, volume and temperature
+
+# Keyword arguments
+
+- `buf::Union{BrusilovskyThermoBuffer, NamedTuple, AbstractDict}`: buffers for intermediate
+calculations
+"""
+function log_c_activity_wj(
+    mix::BrusilovskyEoSMixture{T},
+    nmol::AbstractVector,
+    volume::Real,
+    RT::Real;
+    buf = nothing
+) where {T}
+    nc = ncomponents(mix)
+    log_ca = Vector{T}(undef, nc)
+    jacobian = similar(log_ca, (nc, nc))
+    return log_c_activity_wj!(log_ca, jacobian, mix, nmol, volume, RT; buf = buf)
 end
 
-function log_activity!(
-    mix::ChemPotentialMixture{<:BrusilovskyEoSMixture},
+function log_c_activity_wj!(
+    log_ca::AbstractVector,
+    jacobian::AbstractMatrix,
+    mix::BrusilovskyEoSMixture,
     nmol::AbstractVector,
     volume::Real,
-    RT::Real
+    RT::Real;
+    buf = nothing
 )
-
-    aij = mix.buffers.aij
-    ai = mix.buffers.aux1
-    log_a = mix.buffers.log_a
-    return log_activity!(log_a, ai, aij, mix.mixture_eos, nmol, volume, RT)
+    return __log_c_activity_wj_impl__(log_ca, jacobian, mix, nmol, volume, RT, buf)
 end
 
-function log_activity_wj!(
-    log_a_::AbstractVector,
+function __log_c_activity_wj_impl__(
+    log_ca::AbstractVector,
     jacobian::AbstractMatrix,
-    aij_::AbstractMatrix,
-    aux1::AbstractVector,
-    aux2::AbstractVector,
-    mix::BrusilovskyEoSMixture{T},
+    mix::BrusilovskyEoSMixture,
     nmol::AbstractVector,
     volume::Real,
-    RT::Real
-) where {T}
-    A, B, C, D, aij = eos_parameters!(aij_, aux1, mix, nmol, RT)
+    RT::Real,
+    buf::BrusilovskyThermoBuffer
```

Comment:
Docstring at line 141 says

```julia
 - `buf::Union{BrusilovskyThermoBuffer, NamedTuple, AbstractDict}`: buffers for intermediate calculations
```

but call of `log_activity_wj(mixture, nmol, volume, RT; buf = (;))` with `Dict` or `NamedTuple` `buf` argument will not be dispatched. Only `buf isa BrusilovskyThermoBuffer` is accessible, otherwise `MethodError: no method found`.

Reply:
That remained from an improperly used copy-pasta. Fixed in 9f7f7600d28b4ad77c1f0f108d067e95de300fe3